src/llmq/quorums.cpp

```diff
@@ -301,6 +301,10 @@ void CQuorumManager::UpdatedBlockTip(const CBlockIndex* pindexNew, bool fInitial
 
 void CQuorumManager::CheckQuorumConnections(const Consensus::LLMQParams& llmqParams, const CBlockIndex* pindexNew) const
 {
+    if (!fMasternodeMode && !utils::IsWatchQuorumsEnabled()) {
```

Comment:
should not it be `CQuorumManager::UpdatedBlockTip` instead?

```diff
 void CQuorumManager::UpdatedBlockTip(const CBlockIndex* pindexNew, bool fInitialDownload) const
+    if (!fMasternodeMode && !utils::IsWatchQuorumsEnabled()) return;
     if (!m_mn_sync->IsBlockchainSynced()) {
         return;
     }
```

`mapQuorumData` seems used only for MNs; `CleanupOldQuorumData` and `TriggerQuorumDataRecoveryThreads` both check flags `fMasternodeMode` too.

Reply:
Good catch! 👍 There are indeed more related things to fix. I opened #5745 to address them all.